source/shared/cpp/ObjectModel/MarkDownBlockParser.cpp

```diff
@@ -449,37 +454,40 @@ bool LinkParser::MatchAtLinkTextEnd(std::stringstream& lookahead)
 // link is in form of [txt](url), this method matches '('
 bool LinkParser::MatchAtLinkDestinationStart(std::stringstream& lookahead)
 {
-    // if peeked char is EOF or extended char, this isn't a match
-    if (lookahead.peek() < 0)
-    {
+    // handles [xx](
+    if (lookahead.peek() < 0) {
+        m_parsedResult.AppendParseResult(m_linkTextParsedResult);
         return false;
     }
 
+    // identify where the destination value ends by marking the position
+    // e.g: ([ab()c])()()() end = 7
+    m_parsingCurrentPos = lookahead.tellg();
+    int i = int(m_parsingCurrentPos);
+    char c;
+    while (lookahead.get(c) && m_linkDestinationStart > 0 ) {
```

Comment:
according to doc (https://www.cplusplus.com/reference/istream/istream/get/) the get method should return EOF when no more characters can be read, could you verify the first condition can fail?